casper/src/test/resources/PoSTest.rho

```diff
@@ -463,6 +465,42 @@ match (
                 }
               }
             }
+          } |
+
+          contract getPosBalance(retCh) = {
+            new accBalances, vaultsCh, posVaultBalanceCh in {
+              contract accBalances(@(_, _, vault), @acc, resultCh) = {
+                new balanceCh in {
+                  @vault!("balance", *balanceCh) |
+                  for (@balance <- balanceCh) {
+                    resultCh!(acc + balance)
+                  }
+                }
+              } |
+
+              @posVault!("balance", *posVaultBalanceCh) |
+              for (@posVaultBalance <- posVaultBalanceCh) {
+                @PoS!("getActiveValidatorVaults", *vaultsCh) |
+                for (@vaults <- vaultsCh) {
+                  @ListOps!("fold", vaults, posVaultBalance, *accBalances, *retCh)
+                }
+              }
+            }
+          } |
+
+          contract closeBlock(ackCh) = {
+            new ackCh0, activeValidatorsCh, setBlockData(`rho:test:block:data:set`), blockDataSet in {
+              @PoS!("closeBlock", *ackCh0) |
+              @PoS!("getActiveValidators", *activeValidatorsCh) |
+              for (@ack <- ackCh0; @activeValidators <- activeValidatorsCh) {
+                // After the first "closeBlock" is done, there are active validators
```

Comment:
I don't understand this comment. There should be active validators all the time, not only after the first closeBlock.

Reply:
It's confusing and I removed it.
I guess what I meant is that once `closeBlock` is done, we are good to set the block sender to one of the active validators.